include/libopencm3/pac55xx/can.h

```diff
@@ -0,0 +1,289 @@
+/**
+ * @brief CAN definitions for the Qorvo PAC55xx series of microcontrollers.
+ *
+ * @addtogroup PAC55xx_can CAN
+ * @ingroup PAC55xx_defines
+ * @author Kevin Stefanik <kevin@allocor.tech>
+ * LGPL License Terms @ref lgpl_license
+ * @date February 13, 2020
+ *
+ * Definitions in this file come from the PAC55XX Family User Guide Rev 1.23
+ * by Active-Semi dated November 19, 2019.
+ *
+ * Note: all memory-mapped writes must be performed using 32-bit registers.
+ * Any 8-bit memory-mapped registers below may only be used to read.
```

Comment:
Where is this documented?  I can't find it in the user guide.

Reply:
It's not.  But when I tried to do MMIO8(), things blew up.  Since it's a 32-bit bus, it must be 32-bit transactions.  It's probably a limitation of the peripherals themselves rather than the bus.

Reply:
Anyway, it's not a fun rabbit-hole to go down.  So, I got rid of all MMIO8() references.  Should we get rid of this?  Or is there a better place to put documentation like this, @karlp ?

Reply:
FYI, this was the case in the vendor's SDK, so I did confirm with their support that all write accesses must be 32-bits. 

It might be interesting to make an `MMIOXXRO` macro that declares the register as `const` to enforce read only access?

Reply:
ok then :) yuckkkk.  I was just trying to review things and all the bit offsets are different because it's all documented as just 8 bit registers :)  Let's not block this any longer on this at least.  I'll take it as fact now and keep going

Reply:
Ok, so.... so I'm 100% clear on this, the docs are 8bit, the register is defined as w1c bits, (ie, you can write zeros all day with no impact, you only write 1s to clear the bit, very standard isr register flags, particularly talking abotu CAN_ISR now) and it actually _doesn't work_ unless you do a 32bit write to the neighbouring registers as well? And there's no errata on their website.  That's _really_ how this part behaves?  The 32bit description on RMW applies to SRAM, and it claims it's handled internally as a performance cost, not something you have to handle yourself.

Reply:
Yes, you have to handle it yourself.  This is an actual excerpt from the sample application that they provided (and they confirmed in email you have to do 32-bit word-aligned writes):
`PAC55XX_CAN->ISR_SR_CMR_MR = (PAC55XX_CAN->ISR_SR_CMR_MR & 0x00FFFFFF) | ISR_RI; // Clear RX interrupt`